### DISCUSSION TITLE
TeX の差分シンタックスハイライトの指定言語がおかしい

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#557] [要望・アイデア・不具合]
Qiita のシンタックスハイライトでは、コードブロックの指定言語を `diff_lang` とすることで `lang` に対応した差分表示にすることが出来ます。

TeX の差分シンタックスハイライトに関して、指定言語が `diff_te_x` となっており奇妙です。（`diff_te_x` は TeX の差分シンタックスハイライトとしてきちんと機能しています）
また、`tex` のエイリアスとして `latex` がありますが、`diff_latex` は存在しません。

![image](https://github.com/increments/qiita-discussions/assets/75878492/c3773d91-1982-41b5-bc13-9e2ce7dbd889)

TeX および LaTeX の差分シンタックスハイライトの指定言語に `diff_tex` と `diff_latex` を追加してほしいです。

**Comment by tanaka-asami415** [comment]
@Yarakashi-Kikohshi 
ご報告ありがとうございます🙏
調査の上、対応します！

**Reply by yarakos95** [reply]
@tanaka-asami415 

現在もこの問題は解消されていません。

せかすようで申し訳ないのですが、この問題への対応はどれくらいになるでしょうか。

**Reply by tanaka-asami415** [reply]
@Yarakashi-Kikohshi 
ご不便をおかけして申し訳ありません。
大まかな原因は判明しているのですが、直近の開発優先度から着手できていない状況です🙇
お待たせして大変申し訳ありませんが、お待ちいただけると幸いです。

**Reply by yarakos95** [reply]
気付いたら丸一年だったので、忘れられているのかと心配になってしまいました
対応よろしくお願いします。

**Reply by P-SiZK** [reply]
@yarakos95 
大変お待たせいたしました :bow:
本件について、対応が完了いたしました！

現在は `diff_tex` および `diff_latex` のエイリアスが利用可能になっています。
また今回の対応で、特定の言語に限らず、任意の言語のエイリアスと diff が対応するようになりました。

なお、`diff_te_x` のような以前までの指定については、既存の記事が壊れないよう引き続き動作するように残しております。

よろしければご確認ください :pray:

**Reply by yarakos95** [reply]
反応が遅くなり申し訳ありません。

対応完了ありがとうございます！
実際に試したところ、`diff_tex` および `diff_latex` でも差分シンタックスハイライトが機能することを確認しました 👀 

-----

ところで、試していたときに気付いたのですが、指定言語が大文字か小文字かに関して一貫性が無いように感じます。

基本的に指定言語は case-insensitive ですが、`diff_lang` の `diff` 部分のみ `Diff` のような形にはできません。
以下に一覧としました。

- **`diff`**: case-insensitive

    e.g. `diff` でも `Diff` でも `diFF` でも良い

- **`lang`**: case-insensitive

    e.g. `lang` でも `Lang` でも `laNG` でも良い

- **`diff_lang` の `lang` 部分**: case-insensitive

    e.g. `diff_lang` でも `diff_Lang` でも `diff_laNG` でも良い

- **`diff_lang` の `diff` 部分**: case-sensitive :interrobang:

    e.g. `diff_lang` は良いが、`Diff_lang` や `diFF_lang` はダメ

どれくらいの人がこれに引っ掛かるのか分かりませんが、`diff` が case-insensitive で良いと思っていたら `diff_lang` は部分的に case-sensitive になり得る状況はいただけないように感じます。

**Reply by P-SiZK** [reply]
@yarakos95 
ご確認ありがとうございます！
また、追加の報告もありがとうございます :pray:

ご報告いただいた、`diff` 部分の case-sensitive の件につきましても修正をいたしました。
これで、`Diff_lang` や `diFF_lang` といった指定でも問題なく表示されるようになっております。

こちらも、よろしければご確認ください。

**Reply by yarakos95** [reply]
`diff_lang` でも case-insensitive になっていることも確認しました 👀 

重箱の隅をつつくような指摘であったかなと思いましたが、早速対応していただき ありがとうございます。
